public/usage-examples/graphics/fill_circle_on_bitmap-1-beyond.cpp

```diff
@@ -0,0 +1,115 @@
+#ifdef APPLE
+#include <SDL.h>
+#else
+#include <SDL2/SDL.h>
+#endif
+#include <iostream>
+
+// Function to create an SDL window with error handling
+SDL_Window* sdl_open_window(const char* window_title, const int width, const int height)
+{
+    // Declare Variables
+    SDL_Window* window = nullptr;
+
+    // Check for successful initialisation
+    if (SDL_Init(SDL_INIT_VIDEO) < 0)
+    {
+        std::cout << "SDL could not be initialized: " << SDL_GetError() << std::endl;
+        exit(1);
+    }
+
+    // Create Window
+    window = SDL_CreateWindow(
+        window_title,
+        SDL_WINDOWPOS_CENTERED,
+        SDL_WINDOWPOS_CENTERED,
+        width,
+        height,
+        SDL_WINDOW_OPENGL);
+
+    // Error handling for window
+    if (window == nullptr)
+    {
+        SDL_LogError(SDL_LOG_CATEGORY_ERROR, "Could not create window: %s\n", SDL_GetError());
+        exit(1);
+    }
+
+    return window;
+}
+
+// Function to draw a filled circle using midpoint algorithm
+void draw_filled_circle(SDL_Renderer* renderer, int cx, int cy, int radius, SDL_Color color)
```

Comment:
```suggestion
void draw_filled_circle(SDL_Renderer* renderer, int cx, int cy, int radius, SDL_Color color)
{
    SDL_SetRenderDrawColor(renderer, color.r, color.g, color.b, color.a);
    for (int y = -radius; y <= radius; y++)
    {
        int dx = static_cast<int>(sqrt(radius * radius - y * y));
        SDL_RenderDrawLine(renderer, cx - dx, cy + y, cx + dx, cy + y);
    }
}

```

While the function can be improved using above given code but the code uses a little more maths, but the time complexity and the lines of code are much more improved. It is upto the author of the code to incorporate those changes or not.